nncf/torch/quantization/init_range.py

```diff
@@ -154,50 +165,64 @@ def generate_collectors_for_range_init_statistics_collection(
-            reduction_shape_converted = collector_params.convert_reduction_shape(per_sample_stats=True)
-            return PTMixedMinMaxStatisticCollector(
-                collector_params.use_per_sample_stats(per_sample_stats=True),
-                collector_params.use_abs_max,
-                collector_params.use_means_of_mins,
-                collector_params.use_means_of_maxs,
-                reduction_shape_converted,
-                reduction_shape,
-                num_samples,
+            return get_mixed_min_max_statistic_collector(
+                use_abs_max=collector_params.use_abs_max,
+                use_means_of_mins=collector_params.use_means_of_mins,
+                use_means_of_maxs=collector_params.use_means_of_maxs,
+                num_samples=num_samples,
+                **collector_kwargs,
             )
         if init_config.init_type == "mean_min_max":
-            reduction_shape_converted = collector_params.convert_reduction_shape(per_sample_stats=False)
-            return PTMeanMinMaxStatisticCollector(
-                collector_params.use_per_sample_stats(per_sample_stats=False),
-                collector_params.use_abs_max,
-                reduction_shape_converted,
-                reduction_shape,
-                num_samples,
+            return get_mixed_min_max_statistic_collector(
+                use_abs_max=collector_params.use_abs_max,
+                use_means_of_mins=True,
+                use_means_of_maxs=True,
+                num_samples=num_samples,
+                **collector_kwargs,
             )
         if init_config.init_type == "threesigma":
-            return PTMedianMADStatisticCollector(reduction_shape, num_samples)
+            return get_median_mad_statistic_collector(
+                num_samples=num_samples,
+                **collector_kwargs,
+            )
         if init_config.init_type == "percentile":
             min_percentile = init_config.init_type_specific_params.get("min_percentile", 0.1)
             max_percentile = init_config.init_type_specific_params.get("max_percentile", 99.9)
-            return PTPercentileStatisticCollector([min_percentile, max_percentile], reduction_shape, num_samples)
+            return get_precentile_tensor_collector(
+                percentiles_to_collect=[min_percentile, max_percentile],
+                num_samples=num_samples,
+                **collector_kwargs,
+            )
+
         if init_config.init_type == "mean_percentile":
             min_percentile = init_config.init_type_specific_params.get("min_percentile", 0.1)
             max_percentile = init_config.init_type_specific_params.get("max_percentile", 99.9)
-            return PTMeanPercentileStatisticCollector([min_percentile, max_percentile], reduction_shape, num_samples)
+            return get_mean_percentile_statistic_collector(
+                percentiles_to_collect=[min_percentile, max_percentile],
```

Comment:
Why **per**centile and not **pre**centile? What is the difference while calculation in our code?

Reply:
Covered this, please check